### ISSUE TITLE
"ld-linux-x86-64.so.2  --help" doesn't work in some distros

### ISSUE THREAD

**Issue by gitterdude** [brummer10/guitarix#196]
trunk/waftools/cpu_optimization.py in 0.46+ uses`ld-linux-x86-64.so.2 --help` to determine if the CPU can use x86-64-v3 or not.
This gives the error  _"--help: error while loading shared libraries: --help: cannot open shared object file"_ on some distros.

A more universal way (for Linux anyway) could be to check if /proc/cpuinfo has "avx2" or not (taken from the x86-64-level tool at https://github.com/HenrikBengtsson/x86-64-level/blob/develop/x86-64-level)

**Comment by brummer10** [comment]
Yep, on the Ratatouille project I'm using the cpu flags to get the supported optimisation level, I was just to lazy to implement it here. 
I'll take this issue as a reminder.

**Comment by brummer10** [comment]
This is fixed in the mean time.
